PWGHF/TableProducer/candidateCreatorXicToXiPiPi.cxx

```diff
@@ -0,0 +1,655 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file candidateCreatorXicToXiPiPi.cxx
+/// \brief Reconstruction of Ξc± → (Ξ∓ → (Λ → p π∓) π∓) π± π± candidates
+///
+/// \author Phil Lennart Stahlhut <phil.lennart.stahlhut@cern.ch>, Heidelberg University
+/// \author Jinjoo Seo <jseo@cern.ch>, Heidelberg University
+
+#ifndef HomogeneousField
+#define HomogeneousField
+#endif
+
+#include <KFParticleBase.h>
+#include <KFParticle.h>
+#include <KFPTrack.h>
+#include <KFPVertex.h>
+#include <KFVertex.h>
```

Comment:
Why do we need to support KFParticle?

Reply:
In the Run 2 analysis we saw that the reconstruction of the XicPlus can be significantly improved with the KFParticle using topological and mass constraints. In addition to that, we also have the full covariance matrix of all the particles available throughout the reconstruction workflow such that the treatment of the uncertainties might be overall better.

For the moment, we added it in order to test if we can benefit from it in the Run 3 analysis as well. If not (significantly), we will probably transition fully to the DCAFitter.